### PR TITLE
Prioritize reading node_invalid over error for host re-enrollment

### PR DESCRIPTION
**Pull request by zwass**
A TLS server might like to return `"node_invalid": true` and an error string
indicating why the node was invalid. The old behavior caused `node_invalid` to
be ignored if `error` was specified, therefore making it impossible for
osqueryd to resolve the error.

With the new behavior, osqueryd can now re-enroll when a reason is provided.

### CODE REVIEW

File: osquery/remote/utility.h [hunk 1]
```diff
@@ -114,17 +114,23 @@ class TLSRequestHelper : private boost::noncopyable {
...
         }
-        return Status(1, "Request failed: Invalid node key");
+
+        std::string status = "Request failed: Invalid node key";
+        if (output.count("error") > 0) {
+          status += ": " + output.get("error", "<unknown>");
+        }
+        return Status(1, status);
       }
+    } else if (output.count("error") > 0) {
```
Comment (theopolis) [inline osquery/remote/utility.h]: What if `{node_invalid: false, error: "Something is wrong!"}` is returned?

**Comment by theopolis** [comment]
Thanks @zwass, mind handling the other logic problem highlighted in-line. If we drop the `else` out of the conditions it should be OK.

File: osquery/remote/utility.h (hunk 1 above)
Comment (zwass) [inline osquery/remote/utility.h]: Word. Good catch.

**Comment by osqueryer** [comment]
:-1: The commit 394294cf09e8b2eb944e73d03667563032e09ecf (Job results: [518](https://jenkins.osquery.io/job/osqueryPullRequestBuild/518/)) failed one or more tests (**Windows**).

**Pull request merged**